Source/Xamarin/Prism.Forms/Navigation/Xaml/Navigation.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Windows.Input;
+using Prism.Common;
+using Prism.Ioc;
+using Xamarin.Forms;
+using Xamarin.Forms.Xaml;
+
+namespace Prism.Navigation.Xaml
+{
+    public abstract class Navigation : IMarkupExtension, ICommand
+    {
+        protected internal static readonly BindableProperty RaiseCanExecuteChangedInternalProperty =
+            BindableProperty.CreateAttached("RaiseCanExecuteChangedInternal",
+                typeof(Action),
+                typeof(Navigation),
+                default(Action));
+
+        public static readonly BindableProperty CanNavigateProperty =
+            BindableProperty.CreateAttached("CanNavigate",
+                typeof(bool),
+                typeof(Navigation),
+                true,
+                propertyChanged: OnCanNavigatePropertyChanged);
+
+        protected internal static readonly BindableProperty NavigationServiceProperty =
+            BindableProperty.CreateAttached("NavigationService",
+                typeof(INavigationService),
+                typeof(Navigation),
+                default(INavigationService));
+
+
+        protected BindableObject Bindable;
+        protected IEnumerable<BindableObject> BindableTree;
+        protected bool IsNavigating;
+
+        public bool AllowDoubleTap { get; set; } = false;
```

Comment:
I can't think of any scenario where allowing a double tap for a navigation makes sense. I would remove this and make sure that CanExecute always returns false when IsNavigating is true

Reply:
I was actually just talking to a peer about this one, I was at least looking for a better name

Reply:
In my mind it's about is the command executing... if it is we don't want to execute again until after we've completed... and since it's a navigation away from the given page you shouldn't be able to hit it again unless you navigate back...

Reply:
plus the user can do their own logic bound to `CanNavigate`

Reply:
Done